### PR TITLE
Next update 13

### PR DESCRIPTION
**Pull request by ivanaprebeg**
## Summary
Update Next.js to the version 13.

### Documentation: 
[Next 13](https://nextjs.org/blog/next-13)
[Upgrade guide](https://nextjs.org/docs/upgrading)

## Description
- [x] Node version >= 14.6.0
- [x] React >= 18.2.0
- [x] `next/image` renamed to `next/legacy-image`, however we don't use `next-image` so this didn't affect anything
- [x] `next/link` child can't be `<a>`, run codemod and changed it on several places 

## Check if there is something more to do about:
- [x] The swcMinify configuration property was changed from `false` to true.
- [x] The target configuration property has been removed and superseded by [Output File Tracing](https://nextjs.org/docs/advanced-features/output-file-tracing).

### CODE REVIEW

File: src/components/shared/navigations/MainNavigation/MainNavigation.tsx [hunk 1]
```diff
@@ -50,16 +50,14 @@ export const MainNavigation: FC = () => {
...
-						<Image htmlWidth="64px" src="/images/logo-infinum.png" />
-					</a>
+					<Image htmlWidth="64px" src="/images/logo-infinum.png" />
 				</NextLink>
 
 				<HStack>
-					<NextLink href="/" passHref>
+					<NextLink href="/" passHref legacyBehavior>
 						<NavLink>Home</NavLink>
 					</NextLink>
```
Comment (isBatak) [inline src/components/shared/navigations/MainNavigation/MainNavigation.tsx:59]:
```suggestion
					<NavLink as={NextLink} href="/">Home</NavLink>
```

File: src/components/shared/navigations/MainNavigation/MainNavigation.tsx [hunk 2]
```diff
@@ -50,16 +50,14 @@ export const MainNavigation: FC = () => {
...
-					<NextLink href="/flights" passHref>
+					<NextLink href="/flights" passHref legacyBehavior>
 						<NavLink>Flights</NavLink>
 					</NextLink>
```
Comment (isBatak) [inline src/components/shared/navigations/MainNavigation/MainNavigation.tsx:62]:
```suggestion
					<NavLink as={NextLink} href="/flights">Flights</NavLink>
```

File: src/components/shared/footers/MainFooter/MainFooter.tsx [hunk 3]
```diff
@@ -14,9 +14,7 @@ export const MainFooter = () => (
 			>
 				<Stack align="start" spacing={{ base: '6', md: '8' }}>
 					<NextLink href="/" passHref>
```
Comment (isBatak) [inline src/components/shared/footers/MainFooter/MainFooter.tsx:16]:
```suggestion
					<NextLink href="/">
```

File: src/components/features/login/Login/Login.tsx [hunk 4]
```diff
@@ -11,14 +11,14 @@ export const Login: FC<ILoginProps> = () => {
...
 				<Stack spacing="6">
 					<Stack textAlign="center" spacing={{ base: '2', md: '3' }}>
 						<Heading size="lg">{t('heading')}</Heading>
 						<HStack justify="center" spacing="1">
 							<Text color="muted">{t('registerMessage')}</Text>
-							<NextLink href="/register" passHref>
+							<NextLink href="/register" passHref legacyBehavior>
 								<Button as="a" colorScheme="red" variant="link">
 									{t('signUp.label')}
 								</Button>
```
Comment (isBatak) [inline src/components/features/login/Login/Login.tsx:24]:
```suggestion
							<Button as={NextLink} href="/register" colorScheme="red" variant="link">
```

**Review by kristian240 — DISMISSED** [review]

**Review by isBatak — CHANGES_REQUESTED** [review]

File: src/components/features/login/Login/Login.tsx (hunk 4 above)
Comment (ivanaprebeg) [reply src/components/features/login/Login/Login.tsx:24]: Is this suggestion correct? It seems like label gets deleted

Comment (kristian240) [reply src/components/features/login/Login/Login.tsx:24]: What label?

Comment (ivanaprebeg) [reply src/components/features/login/Login/Login.tsx:24]: `{t('signUp.label')}`

Comment (kristian240) [reply src/components/features/login/Login/Login.tsx:24]: It is not correct 😄 
```suggestion
								<Button as={NextLink} colorScheme="red" variant="link" href="/register">
									{t('signUp.label')}
								</Button>
```

Comment (isBatak) [reply src/components/features/login/Login/Login.tsx:24]: Yes my bad

Comment (kristian240) [reply src/components/features/login/Login/Login.tsx:24]: This is also not correct since there is still a closing tag of NextLink

**Review by isBatak — APPROVED** [review]

**Pull request merged**